pkg/api/customization/cluster/validator.go

```diff
@@ -270,3 +275,72 @@ func (v *Validator) accessTemplate(request *types.APIContext, spec *mgmtclient.C
 
 	return nil
 }
+
+// validateGenericEngineConfig allows for additional validation of clusters that depend on Kontainer Engine or Rancher Machine driver
+func (v *Validator) validateGenericEngineConfig(request *types.APIContext, spec *v3.ClusterSpec) error {
+
+	if request.Method == http.MethodPost {
+		return nil
+	}
+
+	clusterName := request.ID
+	prevCluster, err := v.ClusterLister.Get("", clusterName)
+	if err != nil {
+		return httperror.WrapAPIError(err, httperror.InvalidBodyContent, err.Error())
+	}
+	if spec.AmazonElasticContainerServiceConfig != nil {
+		err := validateEKS(*prevCluster.Spec.GenericEngineConfig, *spec.AmazonElasticContainerServiceConfig)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+
+}
+
+func validateEKS(prevCluster, newCluster map[string]interface{}) error {
+	// check config is for EKS clusters
+	if driver, ok := prevCluster["driverName"]; ok {
+		if driver != service.AmazonElasticContainerServiceDriverName {
+			return nil
+		}
+	}
+
+	// don't allow for updating subnets
+	if prev, ok := prevCluster["subnets"]; ok {
+		if new, ok := newCluster["subnets"]; ok {
+			prevSlice, ok := prev.([]interface{})
+			if !ok {
+				return httperror.NewAPIError(httperror.InvalidBodyContent, "unable to convert subnets to interace slice")
```

Comment:
```suggestion
				return httperror.NewAPIError(httperror.InvalidBodyContent, "unable to convert subnets to interface slice")
```

Reply:
This is a weird case too, that those subnets exist but can't be typecast ? I almost wonder if we should just return nil here, because its not what we expected.

Reply:
I think this provides a useful case for clients and api users. Although I do find it hard to imagine a malformed request will make it this far, since it has to have passed the object conversion steps.

Reply:
In this case it's not the request thats malformed though, its the original cluster. Trying to catch the case where perhaps there is some old cluster out there that never got this type set on it or something.

Reply:
Two parts here.
1. The error is not a useful error to a user. If I got that error what would I do with it? At some point that value was put into an AWS config, which means it was valid and even if it's empty it would get the zero value for the field, an empty array. 
2. I don't think we should block the update if Rancher did something bad in the past with this field.

Reply:
Was there an outcome to this ?

Reply:
I think we decided on not catching errors here and just returning nil